### PR TITLE
Fix discoverability of submodules inside `pd.util`

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: #17166 

This PR fixes the discoverability of the submodules of attributes and modules inside `pd.util`. Somehow `importlib.import_module("pandas.util").__dict__` doesn't display submodules and only root level attributes.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/pandas/_wrappers/pandas.py [hunk 1]
```diff
@@ -76,11 +76,15 @@ def _pandas_util_dir():
...
-        "hash_pandas_object",
         "Appender",
         "Substitution",
+        "_exceptions",
+        "_print_versions",
         "cache_readonly",
+        "capitalize_first_letter",
+        "hash_array",
+        "hash_pandas_object",
+        "version",
```
Comment (Matt711) [inline python/cudf/cudf/pandas/_wrappers/pandas.py:87]: FYI there are some attributes missing. Do we need them?
```suggestion
        "version",
         "_tester",
         "_validators",
```

**Review by Matt711 — APPROVED** [review]

File: python/cudf/cudf/pandas/_wrappers/pandas.py (hunk 1 above)
Comment (Matt711) [reply python/cudf/cudf/pandas/_wrappers/pandas.py:87]: `_decorators` too

**Comment by Matt711** [comment]
The errors I'm seeing are
```
ERROR python/cudf/cudf_pandas_tests/test_array_function.py - AttributeError: module 'pandas.util' has no attribute 'capitalize_first_letter'
ERROR python/cudf/cudf_pandas_tests/test_cudf_pandas.py - AttributeError: module 'pandas.util' has no attribute 'capitalize_first_letter'
ERROR python/cudf/cudf_pandas_tests/test_cudf_pandas_cudf_interop.py - AttributeError: module 'pandas.util' has no attribute 'capitalize_first_letter'
ERROR python/cudf/cudf_pandas_tests/test_cudf_pandas_no_fallback.py - AttributeError: module 'pandas.util' has no attribute 'capitalize_first_letter'
ERROR python/cudf/cudf_pandas_tests/test_profiler.py - AttributeError: module 'pandas.util' has no attribute 'capitalize_first_letter'
```

**Comment by mroeschke** [comment]
Noting that there will be discrepancies depending on pandas 2.x version

pandas 2.2 (and upcoming 2.3) will have `capitalize_first_letter`

pandas 2.0, 2.1 (and upcoming 3.0) will not have `capitalize_first_letter`

**Pull request merged**